byway/src/main/webapp/routepage.js

```diff
@@ -49,27 +63,145 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
   });
 }
 
-function addToStops(stop){
-  console.log("add" + stop.placename + "to the ArrayList in the servlet");
+/** Clear the stops panel in the html */
+function clearStops(){
+  const stopList = document.getElementById('stop-list');
+  if(stopList != null){
+    stopList.innerText = ""; // clear list
+  }
 }
 
-function getRecs() {
-  fetch('/api/recs')
+/** Render stop list  */
+function renderStops(stop, fromDatastore){
+  const stopList = document.getElementById('stop-list');
+  // add to stop array in the js 
+  if(fromDatastore){
+    stops.push(stop);
+  }
+  let btn = document.createElement('button');
+  btn.id = stop.id;
+  btn.innerText = stop.placename;
+  btn.className =  "btn rec-btn";
+  btn.addEventListener("click", function() {
+    deleteFromStops(stop); 
+  });
+  stopList.appendChild(btn);
+}
+
+/** Get the new list of stops from datastore onload */
+function getStopsOnload(){
+  clearStops();
+  fetch('/api/stop')
   .then(response => response.json())
-  .then((recs) => {
-    const recList = document.getElementById('rec-list');
-    recList.innerText = ""; // clear old comments 
-    recs.forEach((stop) => {
-      let btn = document.createElement("button");
-      btn.innerText = stop.placename;
-      btn.setAttribute("class", "btn rec-btn");
-      btn.addEventListener("click", function() {
-        addToStops(stop);
-      });
-      recList.appendChild(btn);
+  .then((stopsResponse) => {
+    stopsResponse.forEach((stop)=>{
+      renderStops(stop, true);
```

Comment:
Just add the `stop` to the list of stops here rather than in the `renderStops()` function. It is not really a part of the render logic anyways, and  you can get rid of the extra boolean param.

Reply:
Oh right, that was an unnecessary complication, thanks for pointing that out.